### PR TITLE
[MM-48105]: Change browse channel modal to be infinite scroll

### LINKED ISSUE #21577: Change browse channel modal to be infinite scroll
**Expected behaviour:** 
The browse channel modal should be changed to infinite scroll using a similar loading indicator as the post list infinite scroll. Threshold will be 30 per list and load more when the scroll passes the threshold.

 

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-48105

### PR DESCRIPTION
**Pull request by gitstart**
Summary
Webapp: Change browse channel modal to be infinite scroll

Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/21577

### DEMO VIDEO

https://www.loom.com/share/910baba4e30841f9b83ab5f17f0ca4b0

### Screenshot
Screenshot 2023-01-24 at 10 18 11

### Release note
```release-note
Changed browse channel modal to be infinite scroll
```

### CODE REVIEW

**Comment by M-ZubairAhmed** [comment]
@gitstart why is the ticket at gitstart?

**Comment by M-ZubairAhmed** [comment]
There is an existing PR already in place https://github.com/mattermost/mattermost-webapp/pull/12019 but it got stale will ask the OP if they like to continue or not. If not then we can not mark this as duplicate

**Comment by AshishDhama** [comment]
Whoever reviews this pr, please don't merge it before https://github.com/mattermost/mattermost-webapp/pull/11262
It's a month-long effort to update this modal and I don't it to be delayed due to these changes.

**Comment by gitstart** [comment]
> Whoever reviews this pr, please don't merge it before #11262 It's a month-long effort to update this modal, and I don't want it to be delayed due to these changes.

We didn't see the PR should we close this PR?

**Comment by AshishDhama** [comment]
@gitstart is it ready for review

**Comment by gitstart** [comment]
> @gitstart is it ready for review

@AshishDhama The PR is ready for review.

**Review by tboulis — COMMENTED** [review]
I left some comments. Will also request a review from the web platform team because I might be missing some context

File: components/searchable_channel_list.jsx [hunk 1]
```diff
@@ -124,18 +118,12 @@ export default class SearchableChannelList extends React.PureComponent {
...
-    }
-
-    previousPage = (e) => {
-        e.preventDefault();
-        this.setState({page: this.state.page - 1});
-        this.channelListScroll.current?.scrollTo({top: 0});
+    nextPage = () => {
+        if (!this.props.loading) {
+            const page = this.state.page + 1;
+            this.props.nextPage(page + 1);
```
[the remark this answers is not in the document]
Comment (gitstart) [reply components/searchable_channel_list.jsx]: It should be `this.state.page + 1`. Thank you.

**Comment by hmhealey** [comment]
Echoing this comment from above since we've had some changes in the reviewers for this PR

> Whoever reviews this pr, please don't merge it before #11262 It's a month-long effort to update this modal, and I don't want it to be delayed due to these changes.

**Comment by hmhealey** [comment]
That other PR was merged, so this one is now no longer blocked on that, however it is unfortunately blocked on our migration to a monorepo (see https://github.com/mattermost/mattermost-server/issues/22420 for more information). Once that's merged, we'll we'll hopefully be able to help move PRs like this over to that repository

**Comment by hmhealey** [comment]
The web app now lives in the `webapp/channels` over in https://github.com/mattermost/mattermost-server. We decided to move it to be a monorepo to make both development and our CI pipelines easier, but it's unfortunately disruptive right now. We've updated [our setup docs](https://developers.mattermost.com/contribute/developer-setup/) and added [some steps for migrating to working with the monorepo](https://developers.mattermost.com/contribute/monorepo-migration-notes/).

Copying the changes over there hopefully shouldn't be too hard. There's a few changes to file paths (such as the code living in a dedicated `src` subdirectory), but most things should look the same as they do now. 

Let us know if you could use any help with this. Hopefully the process is smooth.

**Comment by gitstart** [comment]
@hmhealey this PR has been moved to a monorepo https://github.com/mattermost/mattermost-server/pull/22724 cc @M-ZubairAhmed

**Pull request closed without merging**